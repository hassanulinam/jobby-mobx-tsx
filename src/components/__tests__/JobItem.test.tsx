import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import JobItem from "../JobItem";

const dummyDetails = {
  id: "1234",
  title: "My Title",
  rating: 3.6,
  location: "hyderabad",
  employmentType: "full time",
  packagePerAnnum: "6 LPA",
  jobDescription: "My job description",
};

it("should check whether all the elements are present", () => {
  render(<JobItem details={dummyDetails} />);

  const ratingEl = screen.getByText("3.6");
  expect(ratingEl).toBeInTheDocument();
  screen.getByText(dummyDetails.jobDescription);
  const packageEl = screen.getByTestId("packageEl");
  expect(packageEl).toHaveTextContent(dummyDetails.packagePerAnnum);
  expect(screen.getByTestId("horizontal-rule")).toBeInTheDocument();
});

it("should render correctly", () => {
  const tree = renderer.create(<JobItem details={dummyDetails} />).toJSON();
  expect(tree).toMatchSnapshot();
});
