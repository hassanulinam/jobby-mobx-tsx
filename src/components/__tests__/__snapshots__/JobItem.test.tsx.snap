// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<li
  className="cursor-pointer"
  onClick={[Function]}
>
  <div
    className="job-item-card-container"
  >
    <div
      className="flex-row"
    >
      <img
        alt="company logo"
        className="company-logo-img"
      />
      <div>
        <h1
          className="job-title"
        >
          My Title
        </h1>
        <div
          className="flex-row"
        >
          <svg
            color="#ffff00"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#ffff00",
              }
            }
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
            />
          </svg>
          <p
            className="bold ml-1"
          >
            3.6
          </p>
        </div>
      </div>
    </div>
    <div
      className="flex-sm-col flex-row justify-content-between mb-0 mt-3"
    >
      <div
        className="flex-row flex-sm-col"
      >
        <div
          className="flex-row mr-4"
        >
          <svg
            color="#ffffff"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#ffffff",
              }
            }
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5a2.5 2.5 0 010-5 2.5 2.5 0 010 5z"
            />
          </svg>
          <p
            className="ml-1"
          >
            hyderabad
          </p>
        </div>
        <div
          className="flex-row"
        >
          <svg
            color="#ffffff"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#ffffff",
              }
            }
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.5 1A1.5 1.5 0 0 0 5 2.5V3H1.5A1.5 1.5 0 0 0 0 4.5v1.384l7.614 2.03a1.5 1.5 0 0 0 .772 0L16 5.884V4.5A1.5 1.5 0 0 0 14.5 3H11v-.5A1.5 1.5 0 0 0 9.5 1h-3zm0 1h3a.5.5 0 0 1 .5.5V3H6v-.5a.5.5 0 0 1 .5-.5z"
            />
            <path
              d="M0 12.5A1.5 1.5 0 0 0 1.5 14h13a1.5 1.5 0 0 0 1.5-1.5V6.85L8.129 8.947a.5.5 0 0 1-.258 0L0 6.85v5.65z"
            />
          </svg>
          <p
            className="ml-1"
          >
            full time
          </p>
        </div>
      </div>
      <div>
        <p
          className="job-title"
          data-testid="packageEl"
        >
          6 LPA
        </p>
      </div>
    </div>
    <hr
      data-testid="horizontal-rule"
    />
    <div
      className="job-desc"
    >
      <h1
        className="job-desc-heading"
      >
        Description
      </h1>
      <p
        className="line-gap"
      >
        My job description
      </p>
    </div>
  </div>
</li>
`;
